storage/index.go

```diff
@@ -8,16 +8,11 @@ import (
 )
 
 type index interface {
-	Get(key []byte, atIndex uint64) (index uint64, err error)
-	Range(key, end []byte, atIndex uint64) []kipair
-	Put(key []byte, index uint64)
-	Tombstone(key []byte, index uint64) error
-	Compact(index uint64) map[uint64]struct{}
-}
-
-type kipair struct {
-	index uint64
-	key   []byte
+	Get(key []byte, atRev int64) (rev reversion, err error)
+	Range(key, end []byte, atRev int64) ([][]byte, []reversion)
```

Comment:
it is not kvpair. it might be krpair (key and reversion). but i did not find it quite useful.